tony-core/src/main/java/com/linkedin/tony/ReusablePort.java

```diff
@@ -135,54 +186,35 @@ static ReusablePort create() throws IOException, InterruptedException {
-    //(https://docs.oracle.com/javase/8/docs/api/java/net/Socket.html)?
-    // - Tony's default Java version is 8 and port reuse feature is only available in Java 9+:
-    //   https://docs.oracle.com/javase/9/docs/api/java/net/StandardSocketOptions.html#SO_REUSEPORT.
-    //
+  static ReusablePort create(int port) throws IOException {
     // Why not upgrading Tony to Java 9+ given port reuse is supported in Java 9+?
     // - In Linkedin, as of now(2020/08), only Java 8 and 11 are officially supported, but Java 11
     //   introduces incompatibility with Play version tony-portal
     //   (https://github.com/linkedin/TonY/tree/master/tony-portal) is using. Upgrading Play to a
     //   Java 11-compatible version requires non-trivial amount of effort.
 
     Preconditions.checkArgument(port > 0, "Port must > 0.");
-    final EventLoopGroup bossGroup = new EpollEventLoopGroup();
-    ServerBootstrap b = new ServerBootstrap();
-    ChannelFuture future = null;
 
-    try {
-      b.group(bossGroup)
-          .channel(EpollServerSocketChannel.class)
-          .childHandler(new ChannelInitializer<SocketChannel>() {
-            @Override
-            public void initChannel(SocketChannel ch) {
-            }
-          }).option(EpollChannelOption.SO_REUSEPORT, true)
-          .option(ChannelOption.SO_KEEPALIVE, true);
-
-      // Note it's still slightly possible that another tony processes grab the same port after
-      // {@link #getAvailablePort()}, leading to two tensorflow process using the same port.
-      // So adding an extra port check to reduce the risk.
-      if (isPortAvailable(port)) {
-        // Why not using port 0 here which lets kernel pick an available port?
-        // - Since another tony executor can bind to the same port when port 0 and SO_REUSEPORT are
-        //   used together. See how a port is selected by kernel based on a free-list and socket
-        //   options: https://idea.popcount.org/2014-04-03-bind-before-connect/#port-allocation.
-        future = b.bind(port).await();
-        if (!future.isSuccess()) {
-          throw new BindException("Fail to bind to the port " + port);
-        }
-        return new ReusablePort(bossGroup, future);
-      } else {
-        LOG.info("Port " + port + " is no longer available.");
-        throw new BindException("Fail to bind to the port " + port);
+    String socketBindingProcess = String.format("python %s -p %s -d %s",
+        RESERVE_PORT_SCRIPT_PATH, port, Duration.ofHours(1).getSeconds());
+
+    ProcessBuilder taskProcessBuilder = new ProcessBuilder("bash", "-c", socketBindingProcess);
+    taskProcessBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+    taskProcessBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+    if (isPortAvailable(port)) {
+      LOG.info("Starting process " + socketBindingProcess);
+      Process taskProcess = taskProcessBuilder.start();
```

Comment:
Document the change? That you are launching the python process first, and then waiting for its port-file to be created.

Reply:
thanks, added the comment.